### ISSUE TITLE
Remote training fails with UnhandledPromiseRejectionWarning in log stderr

### ISSUE THREAD

**Issue by drwebb** [microsoft/nni#1396]
**Short summary about the issue/question**:
When I run example trails using remote training service platform trials fail to report metrics, stay running for much longer than they should, things generally don't work.

**Brief what process you are following**: 
I am running on a few machines that I have launched as kubernetes pod. I am not using the kubeflow operator because I don't have kubernetes admin rights, and we don't have a NFS server accessible through the cluster. My work around is to create a kubernetes deployment, and run nni in remote training service mode. I enable pod to pod communication by exposing a service. I can ssh in between containers in the deployment no problem. Running the experiment in local mode works perfectly fine.

Here is my example remote config
```
AuthorName: default
experimentName: example_pytorch_cifar10
trialConcurrency: 3
maxExecDuration: 100h
maxTrialNum: 10
#choice: local, remote, pai
trainingServicePlatform: remote
searchSpacePath: search_space.json
#choice: true, false
useAnnotation: false
tuner:
  #choice: TPE, Random, Anneal, Evolution, BatchTuner, MetisTuner
  #SMAC (SMAC should be installed through nnictl)
  builtinTunerName: TPE
  classArgs:
    #choice: maximize, minimize
    optimize_mode: maximize
trial:
  command: python main.py
  codeDir: /mnt/home/src/nni/examples/trials/cifar10_pytorch
  gpuNum: 1
machineList:
  - ip: 10.42.50.106
    username: user
    passwd: user
    # maxTrialNumPerGpu: 1
    # sshKeyPath: /mnt/home/.ssh/id_nni
  - ip: 10.42.58.36
    username: user
    passwd: user
    # maxTrialNumPerGpu: 1
    # sshKeyPath: /mnt/home/.ssh/id_nni
  - ip: 10.42.53.135
    username: user
    passwd: user
    # maxTrialNumPerGpu: 1
    # sshKeyPath: /mnt/home/.ssh/id_nni
``` 

What the output of webui looks like. 
![image](https://user-images.githubusercontent.com/178769/62250136-fe0e4c80-b3a0-11e9-9e54-aeb0ef073d35.png)

Notice how experiments show "SUCESSFULL" yet no default metric. As mentioned before local training works perfect. After the first few failed "SUCESSFUL" runs, no more runs get scheduled.

Running `nnictl log stderr` show the same error a dozen or so times:
```
... [truncated]

(node:25337) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 104)
(node:25337) UnhandledPromiseRejectionWarning: SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at AnonymousObserver.timer.subscribe [as _onNext] (/home/user/miniconda/envs/py36/nni/training_service/remote_machine/remoteMachineTrainingService.js:331:42)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:25337) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 105)
(node:25337) UnhandledPromiseRejectionWarning: SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at AnonymousObserver.timer.subscribe [as _onNext] (/home/user/miniconda/envs/py36/nni/training_service/remote_machine/remoteMachineTrainingService.js:331:42)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:25337) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 106)
```

contents of nnimanagerLog.log
```
[7/31/2019, 9:29:24 PM] INFO [ 'Datastore initialization done' ]
[7/31/2019, 9:29:24 PM] INFO [ 'Rest server listening on: http://0.0.0.0:8080' ]
[7/31/2019, 9:29:24 PM] INFO [ 'RestServer start' ]
[7/31/2019, 9:29:24 PM] INFO [ 'Construct remote machine training service.' ]
[7/31/2019, 9:29:24 PM] INFO [ 'RestServer base port is 8080' ]
[7/31/2019, 9:29:27 PM] INFO [ 'NNIManager setClusterMetadata, key: machine_list, value: [{"ip":"10.42.50.106","username":"user","passwd":"user","port":22},{"ip":"10.42.58.36","username":"user","passwd":"user","port":22},{"ip":"10.42.53.135","username":"user","passwd":"user","port":22}]' ]
[7/31/2019, 9:29:27 PM] INFO [ 'NNIManager setClusterMetadata, key: trial_config, value: {"command":"python main.py","codeDir":"/mnt/home/src/nni/examples/trials/cifar10_pytorch","gpuNum":1}' ]
[7/31/2019, 9:29:27 PM] INFO [ 'Starting experiment: p4DgRZAv' ]
[7/31/2019, 9:29:27 PM] INFO [ 'Change NNIManager status from: INITIALIZED to: RUNNING' ]
[7/31/2019, 9:29:27 PM] INFO [ 'Add event listeners' ]
[7/31/2019, 9:29:27 PM] INFO [ 'Run remote machine training service.' ]
[7/31/2019, 9:29:27 PM] INFO [ 'RestServer start' ]
[7/31/2019, 9:29:27 PM] INFO [ 'RestServer base port is 8081' ]
[7/31/2019, 9:29:28 PM] INFO [ 'NNIManager received command from dispatcher: ID, ' ]
[7/31/2019, 9:29:28 PM] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 0, "parameter_source": "algorithm", "parameters": {"lr": 0.001, "optimizer": "Adam", "model": "googlenet"}, "parameter_index": 0}' ]
[7/31/2019, 9:29:28 PM] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 1, "parameter_source": "algorithm", "parameters": {"lr": 0.001, "optimizer": "Adamax", "model": "densenet121"}, "parameter_index": 0}' ]
[7/31/2019, 9:29:28 PM] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 2, "parameter_source": "algorithm", "parameters": {"lr": 0.1, "optimizer": "SGD", "model": "mobilenet"}, "parameter_index": 0}' ]
[7/31/2019, 9:29:32 PM] INFO [ 'submitTrialJob: form: {"jobType":"TRIAL","hyperParameters":{"value":"{\\"parameter_id\\": 0, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"lr\\": 0.001, \\"optimizer\\": \\"Adam\\", \\"model\\": \\"googlenet\\"}, \\"parameter_index\\": 0}","index":0}}' ]
[7/31/2019, 9:29:32 PM] INFO [ 'submitTrialJob: form: {"jobType":"TRIAL","hyperParameters":{"value":"{\\"parameter_id\\": 1, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"lr\\": 0.001, \\"optimizer\\": \\"Adamax\\", \\"model\\": \\"densenet121\\"}, \\"parameter_index\\": 0}","index":0}}' ]
[7/31/2019, 9:29:32 PM] INFO [ 'submitTrialJob: form: {"jobType":"TRIAL","hyperParameters":{"value":"{\\"parameter_id\\": 2, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"lr\\": 0.1, \\"optimizer\\": \\"SGD\\", \\"model\\": \\"mobilenet\\"}, \\"parameter_index\\": 0}","index":0}}' ]
[7/31/2019, 9:30:02 PM] ERROR [ 'Update job status exception, error is No such file' ]
[7/31/2019, 9:30:02 PM] INFO [ 'Trial job eIlda status changed from WAITING to RUNNING' ]
[7/31/2019, 9:30:32 PM] INFO [ 'Trial job ZITxb status changed from WAITING to RUNNING' ]
[7/31/2019, 9:37:29 PM] INFO [ 'Trial job eIlda status changed from RUNNING to SUCCEEDED' ]
[7/31/2019, 9:37:29 PM] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 3, "parameter_source": "algorithm", "parameters": {"lr": 0.01, "optimizer": "Adagrad", "model": "senet18"}, "parameter_index": 0}' ]
[7/31/2019, 9:37:34 PM] INFO [ 'submitTrialJob: form: {"jobType":"TRIAL","hyperParameters":{"value":"{\\"parameter_id\\": 3, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"lr\\": 0.01, \\"optimizer\\": \\"Adagrad\\", \\"model\\": \\"senet18\\"}, \\"parameter_index\\": 0}","index":0}}' ]
[7/31/2019, 9:37:49 PM] INFO [ 'Trial job ZITxb status changed from RUNNING to SUCCEEDED' ]
[7/31/2019, 9:37:49 PM] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 4, "parameter_source": "algorithm", "parameters": {"lr": 0.0001, "optimizer": "SGD", "model": "googlenet"}, "parameter_index": 0}' ]
[7/31/2019, 9:37:54 PM] INFO [ 'submitTrialJob: form: {"jobType":"TRIAL","hyperParameters":{"value":"{\\"parameter_id\\": 4, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"lr\\": 0.0001, \\"optimizer\\": \\"SGD\\", \\"model\\": \\"googlenet\\"}, \\"parameter_index\\": 0}","index":0}}' ]
```

**How to reproduce it**: 

Not sure now to reproduce
**nni Environment**:
- nni version: 0.9.1.1
- nni mode(local|pai|remote): remote
- OS: Ubuntu 16.04
- python version: 3.6
- is conda or virtualenv used?: Conda
- is running in docker?: Yes, through kubernetes

**Comment by SparkSnail** [comment]
1.Could you please make sure that your nniManager pod is accessible by other pod? If your IP of nniManager is assigned by yourself, you should set `nniManager` field in your config file, for NNI in remote machine will not reach your nniManager pod when send metrics back. [refer](https://github.com/microsoft/nni/blob/master/docs/en_US/Tutorial/ExperimentConfig.md#configuration-spec)
2. I noticed that you use 1 gpu in your trial, could you please make sure you can use `nvidia-smi` command in your remote container? For nni will use this command to check your gpu usage and schedule machine.

**Comment by drwebb** [comment]
@SparkSnail Thank you for the quick response. I have resolved the issue by disabling a http proxy for addresses inside the cluster. It was not readily apparent from the documentation that we also need http access to the nniManager server from the remote machines. I was assuming that everything trial related was done over ssh and that the webui was just for show. Perhaps we could add something in the docs so someone else doesn't get hit by this one? Closing issue.
